packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,152 @@
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  isChecked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isChecked: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    isChecked,
+    isDisabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+  const theme = React.useContext(ThemeContext);
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback((event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!isChecked, event);
+    }
+  }, [isChecked, onChange]);
+
+  const handleOnKeyDown = (event) => {
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      event.preventDefault();
+      enableFocusStyles(sliderButton.current);
+      if (onChange) {
+        onChange(!isChecked, event);
+      }
+    }
+  };
+
+  const handleOnMouseDown = (event) => {
+    event.preventDefault();
+    removeFocusStyles(sliderButton.current);
+  };
+
+  const handleOnBlur = () => {
+    restoreFocusStyles(sliderButton.current);
+  };
+
+  const labelClassNames = cx([
+    'label-text',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const statusLabelClassNames = cx([
+    'status-label-text',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { 'is-selected': isChecked },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-selected': isChecked },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const statusLabelText = isChecked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  const switchClassNames = classNames(cx(
+    'switch',
+    { 'is-disabled': isDisabled },
+    theme.className,
+  ),
+  customProps.className);
+
+  delete customProps.className;
+  return (
+    <button
+      {...customProps}
+      type="button"
+      id={switchId}
+      disabled={isDisabled}
+      aria-checked={isChecked}
+      role="switch"
+      tabIndex="0"
+      onBlur={handleOnBlur}
+      onClick={handleOnClick}
+      onMouseDown={handleOnMouseDown}
+      onKeyDown={handleOnKeyDown}
+      className={switchClassNames}
+      data-terra-switch-show-focus-styles
+      ref={sliderButton}
+    >
+      <div className={cx('label-container')}>
```

Comment:
Add aria-hidden to this element, it will prevent the conflicting reading of the label.

Reply:
But I was thinking that screen reader should read Label also on switch state change. for example when we have switch with label text like `settings` than VO should read it as `settings Off` on switch state change instead of just reading `switch off` by ignoring label text. Please correct me if my understanding is wrong.

Reply:
This would be a change in tandem with adding the aria-label containing the `labelText`. The goal is to remove the status from being read, as it's currently reading "on label off switch" and "off label on switch", which can confuse the user. You can solve it either by adding an aria-hidden to the `'label-container'` and applying an aria-label containing the `labelText`, or by adding an aria-hidden to the `'status-label-text'`.

Both are valid solutions, I just recommended the former as that I think looks cleaner in code.

Reply:
I missed it While testing with VO. yes it does read status of switch twice. fixed this with commit : 7a753a78afacd00dfeae23bd8298f126c4c4ad61